### ISSUE TITLE
Can't access my subscriptions or watched videos after the 19.60 update

### ISSUE THREAD

**Issue by StephanFS** [yuliskov/SmartTube#2978]
One hour ago, the app started showing unrelated random videos that has nothing to do with my subscriptions or related content.  I tried accessing the Subscriptions sidebar, but it shows nothing. The History sidebar shows this message:

Can't load content. 
1) Check date and time on the device. 
2) Check network connection. 
3) Check your proxy settings. 
4) Try to sign in to your account. 
5) Maybe there is nothing in here.

I tried logging off and logging in again, but the app doesn't load an activation code. I even tried to restart my Firestick, but it didn't work either.

**Comment by drapado** [comment]
Same problem here in my chromecast 4k

**Comment by LadyHaywood** [comment]
Yeah was just about to post the same thing. It acts as if you're not signed in at all and you can't press and hold for the context menu either.

**Comment by UOJt8rASW** [comment]
I have the same problem for several hours.

**Comment by fzoli990** [comment]
I downgraded the app but it didn't help.. maybe a server side change is causing the issue.

**Comment by m-weibel** [comment]
Had the same issue, updated to 19.60 after this to no avail.
I assume Google changed stuff.

**Comment by jordangrodecki** [comment]
downgraded to 19.59 and works fine after re-signing in. Seems to be an issue with 19.60 for me.

**Comment by BirukTes** [comment]
Same issue, Awaiting fix, make it quick. Thanks!

**Comment by thoots99** [comment]
Same issue.  How do you even "downgrade" the app?  Presumably no way within the app, I suppose....

**Comment by mumudog520** [comment]
I logout and tried  to re-login my google account but failed. showed the QR code for scan but no access code for sign in.

**Comment by thoots99** [comment]
OK, I tried "downgrading" to 19.59 -- download the release (presumably SmartTube_stable_19.59_arm64-v8a.apk for my NVIDIA Shield 2015), installed that via sideloading with X-plore, I see the usual "update" box --seems like that ought to do it, but after rebooting everything (Shield, router), still no login, no code, and "About" shows version 19.60.  I really really really have never seen anything about "how do you revert to a previous version," and obviously what I tried didn't work.  So, any tips on what on Earth to do?  Thanks!

**Comment by thoots99** [comment]
So, a bit searching suggests I need to uninstall SmartTube altogether, and then install an old version.  I'm not sure I want to dive into those waters quite yet.  I can report that I have done reboots and cold boots on my NVIDIA Shield box, and rebooted my router, all several times.  SmartTube just won't give me a code that I can plug into YouTube's "activation" page to get signed back in.  And SmartTube sure isn't automatically logging me into YouTube.

**Comment by IndefiniteBen** [comment]
I had the same thing using 19.59 and updated to try and fix it.

For a very temporary fix, search works

**Comment by greatbarrierbeef** [comment]
are there any instructions on how to downgrade the apk on fireOS? I'm having the same problem. I usually use the beta version and I even installed the stable release and the same problem persists. if downgrading works as a temp fix, I'm happy to do that, I just need to know how it's done. presumably typing in the apk version on the Downloader app?

**Comment by thoots99** [comment]
IndefiniteBen - "For a very temporary fix, search works."

Yeah, I suppose it does.  My NVIDIA Shield has a "voice remote."  And it's the "2015" version -- the first one.  I've never even tried using the voice remote function.  Not kidding.  Perhaps this'll be a good reason to give it a shot.

**Comment by linuxstv** [comment]
19.61 Beta also faces this same issue.

**Comment by tashhibbert** [comment]
My Toshiba Fire TV has the same problem. Cannot access any of my subscriptions, history or playlists after the update. 😞😞😞

**Comment by greatbarrierbeef** [comment]
okay my previous comment can be disregarded bc I figured out what I needed to do.

for me, using a Toshiba fire TV, I did the following to get it to work.

1. downloaded the recommended "send files to TV" app that's linked on the front page of the repo (in the README) to my phone and the TV
2. downloaded 19.59 stable APK (you'll need to download the appropriate version for your device) to my phone
3. making sure that my phone and the TV were on the same network, sent the appropriate APK over to the TV (note: if you have stable installed already, you'll want to uninstall it first. I was using beta up til this point so that wasn't needed for me, but there will be a conflict otherwise)
4. installed the stable 19.59 APK to my TV
5. opened the app, went to "sign in," code populated as expected and worked on the activation site for YT

hopefully this helps someone who may be as clueless as I was for getting this resolved for the time being. my account loaded as expected, as well as the subs bar and relevant videos. Just had to redo the sidebar which for me wasn't a huge deal.

**Comment by maykar** [comment]
No subscriptions, no history. Signed out to sign back in to see if that helps, but can no longer sign in as "User code is loading..." Is all that displays next to QR code.

**Comment by Archilyte** [comment]
same problem as @maykar here, and the solution provided by @greatbarrierbeef isnt working either

**Comment by antoninvf** [comment]
Another temporary fix, you can still use remote control/casting to the tv for now. Hope this gets fixed soon.

**Comment by Archilyte** [comment]
im gonna assume this is another attempt of youtube trying to block adblockers, and ublock and revanced have been holding out still, hopefully this holds out too...

**Comment by tashhibbert** [comment]
I've tried downgrading and it hasn't worked. Tried what @greatbarrierbeef said too and that hasn't worked, plus I can't get the files to TV app to find my TV 🤬🤬🤬🤬🤬

**Comment by inMikeRotch** [comment]
Appears to be a "Settings" issue between the latest version and the prior version. I am a long-time user, perhaps 2+ years at this point, and I have never reset my Settings to Default.  I got it working after some trial and error.

# Long Version / Debugging
1. I uninstalled the current version and installed an old stable (v18.23) using the Downloader App
2. Was able to view my subscriptions and could sign in. Alright great. 
3. It then prompted me to update to the latest version (v18.60), which I said Yes. It installed without issue. I could sign in into YouTube, everything working. 

For those looking for prior versions:
See: https://github.com/yuliskov/SmartTube/releases/
 
However, when I restored my previous backup, that is when the problems began. So, I think the quick solution is just to do the following:

# Short Version / Simple 
1. Clear your Data Cache for SmartTube, this will reset all of your settings and any sidebar channels.
2. Re-open SmartTube, which should let you authenticate with Youtube
3. Be Happy!

**Comment by rizoid-25** [comment]
I uninstalled and reinstalled 19.59 which did not allow me to log in. I went as far back as 19.54 and was still unable to login with any version.

**Comment by Vulckain** [comment]
Same problem heure, version 16.38

**Comment by Archilyte** [comment]
@inMikeRotch your solution isnt working here either, smarttube isnt loading in the code that allows me to sign in...

**Comment by Wolf68k** [comment]
Which of the versions is for Fire TV: arm64-v8a.apk, armeabi-v7a.apk or x86.apk?

I have a Toshiba 50C350KU running Fire OS 7.6.6.4

**Comment by mengesb** [comment]
I'm on an NVidia Shield and I've tried logout/login, uninstall/reinstall, cache/data clear, just about everything including rebooting - All I get is `User Code is loading...` and the QR code. I've tried several versions to no avail.

**Comment by greatbarrierbeef** [comment]
@Wolf68k should be `armeabi` version, that's what worked on my Toshiba fire TV

**Comment by greatbarrierbeef** [comment]
> I've tried downgrading and it hasn't worked. Tried what @greatbarrierbeef said too and that hasn't worked, plus I can't get the files to TV app to find my TV 🤬🤬🤬🤬🤬

this might be a dumb question so forgive me if it is, but did you make sure that both your TV and your mobile are connected to the same wifi network? you also need to make sure both the TV and the mobile have the transfer files app downloaded to each

**Comment by ADBeasley** [comment]
Tried the following on my FireTV (**with no luck**):

- Restarting the app from the FireTV menu

- Restarting the FireTV stick

I then tried the older version:

- Uninstall app
- Go into Downloader App
- Select Files
- Find the APK I originally downloaded
- Reinstall this APK
- Restart this version - 18.28 in my case

Still cannot sign in (get a QR Code but no user code) but I can watch videos without adverts so going to be doing a lot of searching for now (or taking a beak from YouTube and reading a few books) :-(

Updating back to the latest version still does not return a user code.
One thing I did note on the original version it thought I was still signed in when I went to the settings / accounts

**Comment by greatbarrierbeef** [comment]
> Tried the following on my FireTV (**with no luck**):
> 
>     * Restarting the app from the FireTV menu
> 
>     * Restarting the FireTV stick
> 
> 
> I then tried the older version:
> 
>     * Uninstall app
> 
>     * Go into Downloader App
> 
>     * Select Files
> 
>     * Find the APK I originally downloaded
> 
>     * Reinstall this APK
> 
>     * Restart this version - 18.28 in my case
> 
> 
> Still cannot sign in (get a QR Code but no user code) but I can watch videos without adverts so going to be doing a lot of searching for now (or taking a beak from YouTube and reading a few books) :-(
> 
> Updating back to the latest version still does not return a user code. One thing I did note on the original version it thought I was still signed in when I went to the settings / accounts

a few questions, forgive me if they're redundant:

1. which version were you using? stable vs beta?
2. prior to reinstalling the app, was it the same type as what you used previously? (meaning, you used the beta, un-installed the beta, the reinstalled a downgraded beta version, and/or stable version, etc)
3. if so, did you try downloading a different version than what you normally use? such as downloading the downgraded stable version if you're usually using beta or visa versa?
4. if you reinstalled the same version again but downgraded, did you first clear the cache and data before uninstalling? sometimes android caches user data even if the app itself is uninstalled. this is done to make reinstallation easier (kind of like Windows registry) but it can also cause issues when you're troubleshooting

**Comment by ADBeasley** [comment]
One thing more to note:

Trying the FireTV official YouTube app and selecting a new account with 'web browser' DOES return a code for the yt.be/activate page

Beginning to think YT is stopping the app from linking to user accounts...

**Comment by jimmyhiggs337** [comment]
> One thing more to note:
> 
> Trying the FireTV official YouTube app and selecting a new account with 'web browser' DOES return a code for the yt.be/activate page
> 
> Beginning to think YT is stopping the app from linking to user accounts...

Either they've changed the API for logging in, or yes they're blocking the app

**Comment by ADBeasley** [comment]
> 1. which version were you using? stable vs beta?
> 2. prior to reinstalling the app, was it the same type as what you used previously? (meaning, you used the beta, un-installed the beta, the reinstalled a downgraded beta version, and/or stable version, etc)
> 3. if so, did you try downloading a different version than what you normally use? such as downloading the downgraded stable version if you're usually using beta or visa versa?
> 4. if you reinstalled the same version again but downgraded, did you first clear the cache and data before uninstalling? sometimes android caches user data even if the app itself is uninstalled. this is done to make reinstallation easier (kind of like Windows registry) but it can also cause issues when you're troubleshooting

1) Latest stable version 19.60 - update to this yesterday IIRC
2) Only ever run the stable versions (its on the family TV so no playing allowed)
3) The downgraded version was cached on the FireTV from the original install so I did not download a new version
4) I only uninstalled then reinstalled - never thought about clearing the cache

So I've now tried:

- Clearing all data using the FireTV app control
- Deleting the 19.60 version
- Reinstalling 18.28
- Still get the 'User Code is loading...' message when I try to log in
- Updating to 19.62 and trying to log in still gives me the 'User Code is loading...' message

**Comment by Joomsy** [comment]
@inMikeRotch Neither of those things worked on this end, unfortunately. I cleared cache and data, which didn't work. So, I un-installed, and tried downgrading to 18.23, but it never generates the user code. YouTube's obviously changed something in the authentication API. I've been noticing that STN has been losing connection to my account over the past week or so, but flipping through different sections always fixed it. Seems to be totally broken now, though.

**Comment by ADBeasley** [comment]
@Joomsy - not seen any disconnects on my FireTV here in the UK.

Now getting a message on 19.62:

> Friends, please be patient. I'm investigating the current situation when the app not working. Sorry for the inconvenience.

It's gone 01:20 here so I'm not sure I can help debug anything more as I need to be up early :-(

**Comment by Wolf68k** [comment]
There's a 19.62?

**Comment by ADBeasley** [comment]
**My Bad - I'm on 19.60** (no idea where the 2 came from - its late here and time for my beddibyes)

**Comment by thoots99** [comment]
> > Friends, please be patient. I'm investigating the current situation when the app not working. Sorry for the inconvenience.

This is why I'm not in a big hurry to wipe out my precious settings in any effort to do a work-around.  I will be patient.  :D

**Comment by Spaceboy60** [comment]
I have both the stable and beta versions on my fire stick and I can't sign in with either

**Comment by ADBeasley** [comment]
> This is why I'm not in a big hurry to wipe out my precious settings in any effort to do a work-around. I will be patient. :D

I do not mind trying something this simple - the app is non-working and my config / video history etc is basically a log-on away when fixed and may help debugging the issue.

**Comment by Joomsy** [comment]
@ADBeasley May be something rolling out. YouTube, and by extension Google, slowly drip feeds changes like this so they can try to isolate issues to a small enough crowd prior to it hitting the general masses. This way they can minimize impact, and try to push out a fix if any problems occur. I'm wondering if they're doing it by location/IP allocation, seeing as how not everyone seems to be affected, or affected in the same way.

**Comment by Wolf68k** [comment]
For a moment, when I installed 19.61 beta I was able to get a code but I noticed things still didn't look right. I had backed up my settings before doing anything. When I restored them the I was back to not actually being connected and not able to get a code.

I've tried everything. Clearing the cache and data, uninstall/reinstall. Different versions both beta and release. Nothing. Short of factory resetting the TV and trying again, which I don't think that would do anything.

**Comment by inMikeRotch** [comment]
I was unable to replicate my prior steps. So for now, I have reverted to 18.15 which gives me the code to activate. The one picture is blurry, showing the version sorry.

![Smart-Code-IMG_20231031_202137811](https://github.com/yuliskov/SmartTube/assets/149546149/027028b5-92a5-4ff4-8cbf-8feff70d69fe)
![Smart-v18-15-IMG_20231031_202335115](https://github.com/yuliskov/SmartTube/assets/149546149/bf924a30-b3ec-44c7-9e1d-c9cb37d664b4)

**Comment by cmjdiff** [comment]
The 19.62 update doesn't seem to want to install for me.  When I click "OPEN" on the app installer when it's done, in Settings > About it tells me it's still 19.60

**Comment by devskiski** [comment]
I am currently having an issue we're it says that there is an update for version 19.62 and when I install the update, it says that I still have an update and that I am still on version 19.60 even after applying the 19.62 update over and over again

**Comment by devskiski** [comment]
> The 19.62 update doesn't seem to want to install for me. When I click "OPEN" on the app installer when it's done, in Settings > About it tells me it's still 19.60

I am having the exact same issue

**Comment by Joomsy** [comment]
@cmjdiff It's just a PSA. There's no other way to push notifications through the app. It's just to let you know that the issue is known.

@inMikeRotch I'll give that version a try, and report back.

**Comment by happyguy1** [comment]
So, I run both Smarttube Beta and Smarttube regular on my sonia bravia (android) - i keep Smarttube Beta updated and almost never launch regular smarttube - Everything worked fine today as of ~1:30pm PT - but trying ~5:30PM, i get the exact same symptoms that everyone here has reported - but what i can add is that this occurs on both versions, a fully updated (as of today) STB as well as ST regular, which i used last week but haven't updated for months.  So imo, it is NOT a function of a recent update version.

To describe my symptoms, i am clearly logged in.  I get suggested videos, but they are irrelevant.  I show to have no channels and no subscriptions (though i actually have 50+, as confirmed by checking directly with my youtube account.)  My STB (and ST) history is empty, with a prompt to sign in to "youtube.com/activate" with a displayed QR code (which i won't try) 

In Youtube, all my subscriptions are there, as is my history.

In STB, i can enable subscribe button (turning red) on a channel, but that channel does not display in my STB channel list. And no videos watched as of this evening show up in my STB history.

I agree with other posters that this may be related to google's attempt to deactivate ad blockers - hopefully the STB dev(s) (bless their soul!) will find a workaround!

**Comment by craigmoyle** [comment]
Reminds me that I need to make another donation

**Comment by Joomsy** [comment]
@inMikeRotch Yup, can confirm that the stable release of 18.15 generates a code. Great find. Let's see if it survives an update now.

**Comment by inMikeRotch** [comment]
Just happens to be the last version I manually downloaded. I normally just update from within the App.

**Comment by Besstheone** [comment]
> @inMikeRotch Yup, can confirm that the stable release of 18.15 generates a code. Great find. Let's see if it survives an update now.

Didn't ☹️

**Comment by Archilyte** [comment]
The beta version just got an update...

**Comment by rizoid-25** [comment]
18.15 gave me a code but attempting to activate in Firefox gave me an error. Chrome worked though. I'll Check out the new beta update tomorrow morning.

**Comment by inMikeRotch** [comment]
Updating it will make it revert to broken. I am only running 18.15 for now until a bug fix is released.

**Comment by GrafVonKrolock** [comment]
> I was unable to replicate my prior steps. So for now, I have reverted to 18.15 which gives me the code to activate. The one picture is blurry, showing the version sorry.
> 
> ![Smart-Code-IMG_20231031_202137811](https://user-images.githubusercontent.com/149546149/279552878-027028b5-92a5-4ff4-8cbf-8feff70d69fe.jpg) ![Smart-v18-15-IMG_20231031_202335115](https://user-images.githubusercontent.com/149546149/279552880-bf924a30-b3ec-44c7-9e1d-c9cb37d664b4.jpg)
> 
> I am using this version: https://github.com/yuliskov/SmartTube/releases/tag/18.15s https://github.com/yuliskov/SmartTube/releases/download/18.15s/STubeNext_ststable_v18.15_r_armeabi-v7a.apk

Thanks for the find.
Was having the same sign in issue after updating to the 19.60 version (was working until I updated) and downgrading it to 19.59 wasn't doing anything to fix it. This 18.15 worked on my Nvidia Shield.

**Comment by Wolf68k** [comment]
18.15 works for me. I also dared to use the Restore settings after getting the code and making sure that worked. I seem to still be logged in.

**Comment by craigmoyle** [comment]
And fixed in 19.63 - great work as usual devs

**Comment by mabba18** [comment]
On Nvidia Shield, 19.63 fixed the issue😁

**Comment by weezedog** [comment]
19.63 just released thar restored functionality for me

**Comment by craigmoyle** [comment]
> On Nvidia Shield, 19.63 fixed the issue😁
> 
> Now I have to redo all my settings, because I tried to fix instead of being patient.

Same, probably a good thing to revisit settings anyway as it's been a while

**Comment by Joomsy** [comment]
I certainly needed to flush my settings, so this ended up being beneficial. I've had a list view bug for the past month that I've not bothered to do anything about, and resetting fixed that. So, silver lining to everything, I suppose. Thanks for the quick fix!

**Comment by a32guy** [comment]
Thanks for the quick fix in 19.63 devs!

**Comment by inMikeRotch** [comment]
> @inMikeRotch how many other versions did you try before you landed on this one?
I only had one version, and then I was going to give up. I see they posted a new version which fixes the issue. I haven't tried it yet, will walk back into the living room and see.

**Comment by ginknocab** [comment]
Awesome work devs

**Comment by Exuro-NL** [comment]
I just wanted to say that I'm impressed by how well this app has functioned for me all this time and how appreciative of how good the support is when it comes to these sorts of matters.

**Comment by mengesb** [comment]
Confirmed 19.63 resolved issues with NVidia Shield TV

**Comment by happyguy1** [comment]
YES - FIXED!!

Wanted to donate but there are only 2 options - www.ko-fi/smarttube which purportedly allows a one time donation, but i can't get it to work, and www.patreon.com/smarttube which seems to only allow monthly subscriptions - anyone have a link or know how to access the donation page for smarttube on ko-fi?

all hail yuliskov!!

**Comment by salazr** [comment]
very nice work there @yuliskov, i see the commits and must have been challenging to figure out those tweaks to the auth so quick.

**Comment by buthig666** [comment]
thank you

**Comment by ArgGrr** [comment]
Working great here.

**Comment by thoots99** [comment]
Thanks everyone involved with the quick fix to this issue.  Much appreciated!

**Comment by buthig666** [comment]
@happyguy1 Same!  Was going to donate 5 dollars but the ko-fi is not working.  I wish i had the funds to donate monthly but I simply can't.

**Comment by tashhibbert** [comment]
Yay, all working again. Keep up the good work on his great app 👍👍👍

**Comment by M4TRIX04** [comment]
Thank you, working again, problems fixed.

**Comment by mattm3** [comment]
Came here to check if anyone else was having the same issue as me. I see that it has been updated in version 19.63 but when I click on check for the update I just get the loading icon and nothing else.

I'm using an Nvidia Shield.

**Comment by jimmyhiggs337** [comment]
> Came here to check if anyone else was having the same issue as me. I see that it has been updated in version 19.63 but when I click on check for the update I just get the loading icon and eventually it returns an error that says "Update error"
> 
> I'm using an Nvidia Shield.

I would recommend uninstalling the app and manually installing the 19.63 apk

**Comment by King-Ultra** [comment]
Hi, I've uninstalled the old version and attempt to install 19.63 but it says App not installed.
I'm using Downloader on a Sony Bravia.
Pls help.

**Comment by Joomsy** [comment]
> Hi, I've uninstalled the old version and attempt to install 19.63 but it says App not installed. I'm using Downloader on a Sony Bravia. Pls help.

Sounds like you're downloading the wrong architecture. One APK is 32bit (arm7), and the other is 64bit (arm64-v8). Be sure you're getting the right one for your device. You likely need the 32bit build, though.

**Comment by King-Ultra** [comment]
> > Hi, I've uninstalled the old version and attempt to install 19.63 but it says App not installed. I'm using Downloader on a Sony Bravia. Pls help.
> 
> Sounds like you're downloading the wrong architecture. One APK is 32bit (arm7), and the other is 64bit (arm64-v8). Be sure you're getting the right one for your device. You likely need the 32bit build, though.

You're absolutely right, needed arm 32bit version. Thanks.

**Comment by Xmetalfanx** [comment]
I know this is an old post but just having a similar issue today ... tried the newest 20.05 (if i am right) version and randomly an 18.12 version i think and it says cant load content and when i go to login it cant display the code.   Up until like 2 or so days ago i have had no problems ... even on fresh installs ... will see if maybe there is a SmartTube release (beta channell) above 20.05

**Comment by happyguy1** [comment]
just fyi, i have not experienced the reported (or any other) issue with smarttube or STB throughout today, as recently as 5 minutes ago - i am on v19.78

**Comment by Xmetalfanx** [comment]
deleted my last post related to NewPipe working with a test apk ... in the process of messing around i did turn Rethink (Firewall from F-droid store) and both the 20.05 and 20.14 version of this app are working again (I HAD checked the smarttube rules and everything was set to allow before... .BUT this may have been user error ... at least for me... so i removed my last post

**Comment by yuliskov** [comment]
Got it. Thanks, friend.
